lib/rethinkdb_ecto/repo.ex

```diff
@@ -107,6 +111,28 @@ defmodule RethinkDB.Ecto.Repo do
     end
   end
 
+  defp convert_date({attr, %Ecto.Date{year: year, month: month, day: day}}) do
+    {attr, RethinkDB.Query.time(year, month, day, "Z")}
+  end
+
+  defp convert_date({attr, %Ecto.DateTime{year: year, month: month, day: day, hour: hour, min: min, sec: sec, usec: usec}}) do
+    {attr, RethinkDB.Query.time(year, month, day, hour, min, sec, "Z")}
+  end
+
+  defp convert_date(%RethinkDB.Pseudotypes.Time{epoch_time: epoch_time}) do
```

Comment:
Don't love this as much.

`convert_date` can accept multiple data types, but returning various data types is a bit rough. Perhaps `convert_date_to_ecto` and `convert_date_to_rethinkdb`